src/engine/strat_engine/engine.rs

```diff
@@ -65,25 +73,41 @@ impl StratEngine {
             return Err(EngineError::Engine(ErrorEnum::Error, err_msg));
         }
 
-        let pools = find_all()?;
-
-        let mut table = Table::default();
-        for (pool_uuid, devices) in &pools {
-            let evicted = table.insert(StratPool::setup(*pool_uuid, devices)?);
-            if !evicted.is_empty() {
-
-                // TODO: update state machine on failure.
-                let _ = teardown_pools(table.empty());
+        let mut engine = StratEngine {
+            pools: Table::default(),
+            holding_pool: HashMap::new(),
+            watched_dev_last_event_nrs: HashMap::new(),
+        };
 
-                let err_msg = "found two pools with the same id or name";
-                return Err(EngineError::Engine(ErrorEnum::Invalid, err_msg.into()));
+        match find_all() {
+            Ok(pools) => {
+                for (pool_uuid, devices) in &pools {
+                    match StratPool::setup(*pool_uuid, devices) {
+                        Ok(pool) => {
+                            let evicted = engine.pools.insert(pool);
+                            if !evicted.is_empty() {
+                                // TODO: update state machine on failure.
+                                let _ = teardown_pools(engine.pools.empty());
+                                let err_msg = "found two pools with the same id or name";
+                                return Err(EngineError::Engine(ErrorEnum::Invalid, err_msg.into()));
+                            }
+                        }
+                        Err(e) => {
+                            // Unable to setup pool, it might become complete later?
+                            engine.holding_pool.insert(*pool_uuid, devices.clone());
+                            warn!("Error while doing a pool::setup for pool uuid {} {:?}, \
```

Comment:
Please skip the ```warn!```. It might be useful to shove the error into the ```incomplete_pools``` (```holding_pool```) map along with the rest.